### PR TITLE
MRG: purge old scikits.learn imports

### PR DESCRIPTION
**Pull request by drammock**
Since we've bumped the minimum sklearn to 0.19.1, there should be no need to try/except importing from `scikits.learn` anymore.

### CODE REVIEW

File: mne/utils/docs.py [hunk 1]
```diff
@@ -1199,9 +1199,6 @@ class deprecated(object):
     # Adapted from http://wiki.python.org/moin/PythonDecoratorLibrary,
     # but with many changes.
 
-    # scikit-learn will not import on all platforms b/c it can be
-    # sklearn or scikits.learn, so a self-contained example is used above
-
```
Comment (drammock) [inline mne/utils/docs.py]: I can't figure out what this note was referring to, so I just removed it.  If someone knows, I can restore it if necessary.

File: doc/changes/latest.inc [hunk 2]
```diff
@@ -305,7 +305,7 @@ API
   - NumPy 1.13.3
   - SciPy 1.0.0
   - matplotlib 2.1
-  - scikit-learn 1.19.1 (optional requirement)
+  - scikit-learn 0.19.1 (optional requirement)
```
Comment (drammock) [inline doc/changes/latest.inc]: this looks like just a typo

File: mne/stats/tests/test_cluster_level.py [hunk 3]
```diff
@@ -275,10 +268,7 @@ def test_cluster_permutation_t_test(numba_conditional, stat_fun):
 def test_cluster_permutation_with_connectivity(numba_conditional):
     """Test cluster level permutations with connectivity matrix."""
     try:
-        try:
-            from sklearn.feature_extraction.image import grid_to_graph
-        except ImportError:
-            from scikits.learn.feature_extraction.image import grid_to_graph
+        from sklearn.feature_extraction.image import grid_to_graph
```
Comment (larsoner) [inline mne/stats/tests/test_cluster_level.py]: This `try/except` is old and ugly. Better to just directly do:
```
    from sklearn.feature_extraction.image import grid_to_graph
```
and decorate each function that needs it with
```
@requires_sklearn
```

Comment (drammock) [reply mne/stats/tests/test_cluster_level.py]: I wondered about thata. fixed now.

**Comment by drammock** [comment]
@larsoner this one is ready and already has +1 from alex

**Comment by larsoner** [comment]
Thanks @drammock

**Pull request merged**